test/functionalTest/cases/3190_notify_only_attributes_that_change/notify_only_attributes_that_change_errors.test

```diff
@@ -0,0 +1,187 @@
+payload='{
+  "description": "A subscription to get info about Room1",
+  "subject": {
+    "entities": [
+      {
+        "id": "Room1",
+        "type": "Room"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "A","B","C"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+	"attrs" : [
+	  "A","B"
+	],
+	"exceptAttrs" : [
+	  "C"
+	],
+    "onlyChangedAttrs":true
+  },
+  "expires": "2040-01-01T14:00:00.00Z",
+  "throttling": 5
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo '02. POST /v2/subscriptions, create subscription with onlyChangedAttrs.'
+echo '======================================================================'
+payload='{
+  "description": "A subscription to get info about Room1",
+  "subject": {
+    "entities": [
+      {
+        "id": "Room1",
+        "type": "Room"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "A","B","C"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "onlyChangedAttrs":true
+  },
+  "expires": "2040-01-01T14:00:00.00Z",
+  "throttling": 5
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo '03. POST /v2/subscriptions, create subscription with attrs and onlyChangedAttrs set to incorrect value.'
+echo '======================================================================================================='
+payload='{
+  "description": "A subscription to get info about Room1",
+  "subject": {
+    "entities": [
+      {
+        "id": "Room1",
+        "type": "Room"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "A","B","C"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+	"attrs" : [
+	  "A","B"
+	],
+	"onlyChangedAttrs": "foo"
+  },
+  "expires": "2040-01-01T14:00:00.00Z",
+  "throttling": 5
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+--REGEXPECT--
+01. POST /v2/subscriptions, create subscription with attrs, exceptAttrs and onlyChangedAttrs.
+=============================================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 99
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "http notification has attrs onlyChangedAttrs and exceptAttrs", 
+    "error": "BadRequest"
+}
+
+
+02. POST /v2/subscriptions, create subscription with onlyChangedAttrs.
+======================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 111
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "http notification has onlyChangedAttrs and neither attrs nor exceptAttrs", 
```

Comment:
This usage is legal. If attrs is ommited, the atrrs: [ ] is assumed. As in the previous, just reverting the block in L499-L507 in src/lib/jsonParseV2/parseSubscription.cpp to its original code probably would fix this.

Reply:
fixed in c089631